### PR TITLE
added ucsborganization entity, repository, controller, and tests

### PR DESCRIPTION
**Pull request by rohankumar2025**

### CODE REVIEW

File: src/main/resources/db/migration/changes/UCSBDiningCommons.json
Comment (kylebkumar) [inline src/main/resources/db/migration/changes/UCSBDiningCommons.json:1]: Did you mean to change this file?

Comment (rohankumar2025) [reply src/main/resources/db/migration/changes/UCSBDiningCommons.json:1]: Nah I accidentally changed it when I was trying to change UCSBOrganization.json but I made sure to copy-paste it from the one on main, so there shouldn't be any conflicts.

Comment (BK1031) [reply src/main/resources/db/migration/changes/UCSBDiningCommons.json:1]: It's fine the file is still valid json so we're good to merge.

**Review by BK1031 — APPROVED** [review]
lgtm

**Pull request closed without merging**